datafusion/expr/src/utils.rs

```diff
@@ -1260,6 +1260,24 @@ pub fn collect_subquery_cols(
     })
 }
 
+#[macro_export]
+macro_rules! udf_equals_hash {
```

Comment:
Is this macro needed if we proceed with the proposal in https://github.com/apache/datafusion/issues/16677#issuecomment-3092338265 ?

It seems like that proposal is pretty awesome -- is it possible to do incrementally, or does it require one massive PR ?